### ISSUE TITLE
Docker demo with Grafana shows no data, but connects properly

### ISSUE THREAD

**Issue by exocode** [absmach/magistrala#627]
**BUG REPORT**

## 1. What were you trying to achieve?
Spin up a demo instance via docker compose
Following this tutorial:
https://medium.com/mainflux-iot-platform/mainflux-open-source-iot-platform-set-up-and-usage-70bed698791a and the manual https://mainflux.readthedocs.io/en/latest/storage/#writers

## 2. What are the expected results?
Showing results (data points) from influxdb within Grafana

## 3. What are the received results?
"No data points" within Grafana

## 4. What are the steps to reproduce the issue?
- `git clone https://github.com/mainflux/mainflux.git`
- `cd mainflux`
- `docker-compose -f docker/docker-compose.yml -f docker/addons/influxdb-reader/docker-compose.yml -f docker/addons/influxdb-writer/docker-compose.yml up`
- `mainflux-cli provision test
- `mainflux-cli things connect <YOUR_THINGS_AND_CHANNELS HERE>`
- generate some dummy data
- login into grafana (:3001)
- connect influxdb
- configure like this
<img width="1195" alt="bildschirmfoto 2019-03-04 um 22 11 25" src="https://user-images.githubusercontent.com/1597621/53763292-adb2c180-3eca-11e9-9de9-a277a3548415.png">

## 5. In what environment did you encounter the issue?
MacOS 10.13.6 (17G4015)
Docker: 
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.37 (downgraded from 1.39)
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.05.0-ce
  API version:      1.37 (minimum version 1.12)
  Go version:       go1.10.1
  Git commit:       f150324
  Built:            Wed May  9 22:20:42 2018
  OS/Arch:          linux/amd64
  Experimental:     false
## 6. Additional information you deem important:
First I thought it's an issue that I had no messages, but when I read the values I get a response:

```
curl -s -S -i  -H "Authorization: de03048c-8926-4ec7-9a74-48c973db1641" http://192.168.99.100:8905/channels/ba5ba717-9ba6-4b45-baa0-aa0655e66fc6/messages
HTTP/1.1 200 OK
Content-Type: application/json
Date: Mon, 04 Mar 2019 21:10:12 GMT
Content-Length: 1132

{"messages":[{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"some-base-name:voltage","unit":"V","time":1276020076,"value":120.1},{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"some-base-name:current","unit":"A","time":1276020072,"value":1.3},{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"some-base-name:current","unit":"A","time":1276020071,"value":1.2},{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"name:temperature","unit":"C","time":50,"value":22.3},{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"name:voltage","unit":"V","time":48,"value":11.4},{"channel":"ba5ba717-9ba6-4b45-baa0-aa0655e66fc6","publisher":"c45f1f2f-1d7d-4e13-90a7-bbbd768191aa","protocol":"http","name":"name:temperature","unit":"C","time":48,"value":22.3}]}
```

**But grafana keeps empty** , did I miss some small piece? (I also did try different Grafana versions by modifiying the docker-compose.yml without any success.

Any suggestions? Is this a Grafana issue? A reader issue or a config thing?

Thank you in advance

**Comment by drasko** [comment]
@exocode can you please check if the messages are correctly written in InfluxDB using Influx CLI?

**Comment by drasko** [comment]
Also, you can use these instructions: https://medium.com/mainflux-iot-platform/mainflux-open-source-iot-platform-set-up-and-usage-70bed698791a

**Comment by exocode** [comment]
I had tried your suggested instructions as well (before I posted that issue). BTW: `docker/addons/influxdb/docker-compose.yml` does not exist anymore (mentioned in the blog post).

This is the result of my queries within `influx-cli` (`influx`)
For me it's looking "correct":

```
> SELECT "value" FROM "autogen"."messages" WHERE ("channel" = 'ba5ba717-9ba6-4b45-baa0-aa0655e66fc6')
name: messages
time                value
----                -----
48000000000         22.3
48000000000         11.4
50000000000         22.3
1276020071000000000 1.2
1276020071000000000 1.2
1276020072000000000 1.3
1276020072000000000 1.3
1276020073000000000 1.4
1276020074000000000 1.5
1276020075000000000 1.6
1276020076000000000 120.1
1276020076000000000 1.7
1276020076000000000 1120.1
> SELECT "value" FROM "autogen"."messages"
name: messages
time                value
----                -----
48000000000         22.3
48000000000         11.4
50000000000         22.3
1276020071000000000 1.2
1276020071000000000 1.2
1276020072000000000 1.3
1276020072000000000 1.3
1276020073000000000 1.4
1276020074000000000 1.5
1276020075000000000 1.6
1276020076000000000 120.1
1276020076000000000 1.7
1276020076000000000 1120.1
> SELECT * FROM "autogen"."messages"
name: messages
time                channel                              link name                               protocol publisher                            unit updateTime value
----                -------                              ---- ----                               -------- ---------                            ---- ---------- -----
48000000000         ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      name:temperature                   http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa C    0          22.3
48000000000         ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      name:voltage                       http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa V    0          11.4
50000000000         ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      name:temperature                   http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa C    0          22.3
1276020071000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      some-base-name:current             http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.2
1276020071000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.2
1276020072000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      some-base-name:current             http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.3
1276020072000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.3
1276020073000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.4
1276020074000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.5
1276020075000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.6
1276020076000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      some-base-name:voltage             http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa V    0          1120.1
1276020076000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:current http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa A    0          1.7
1276020076000000000 ba5ba717-9ba6-4b45-baa0-aa0655e66fc6      urn:dev:ow:10e2073a0108006:voltage http     c45f1f2f-1d7d-4e13-90a7-bbbd768191aa V    0          120.1


```

**Comment by exocode** [comment]
Ok I found the problem:
My missing knowledge of InfluxDB did not recognized the timestamp used in InfluxDB.
So the dates provided in all examples I found are using dates below 7 years ago. I had to expand the bandwith 10-20 years ago. :-) And: badaboom I got my two points within this new range:

<img width="1552" alt="bildschirmfoto 2019-03-05 um 17 51 20" src="https://user-images.githubusercontent.com/1597621/53822482-1d2ebc80-3f70-11e9-8979-f8d427f78813.png">


So, thank you very much for pointing me to the right direction!
